sdk/storage/Azure.Storage.Blobs/tests/BlockBlobClientTests.cs

```diff
@@ -2718,6 +2720,426 @@ public async Task OpenWriteAsync_AccessConditionsFail()
             }
         }
 
+        [Test]
+        [ServiceVersion(Min = BlobClientOptions.ServiceVersion.V2020_04_08)]
+        public async Task UploadFromUriAsync()
+        {
+            // Arrange
+            var constants = new TestConstants(this);
+            await using DisposingContainer test = await GetTestContainerAsync();
+            BlockBlobClient sourceBlob = InstrumentClient(test.Container.GetBlockBlobClient(GetNewBlobName()));
+            BlockBlobClient destBlob = InstrumentClient(test.Container.GetBlockBlobClient(GetNewBlobName()));
+
+            // Upload data to source blob
+            byte[] data = GetRandomBuffer(Constants.KB);
+            using Stream stream = new MemoryStream(data);
+
+            BlobHttpHeaders blobHttpHeaders = new BlobHttpHeaders
+            {
+                CacheControl = constants.CacheControl,
+                ContentDisposition = constants.ContentDisposition,
+                ContentEncoding = constants.ContentEncoding,
+                ContentLanguage = constants.ContentLanguage,
+                ContentType = constants.ContentType
+            };
+
+            Metadata metadata = BuildMetadata();
+
+            BlobUploadOptions uploadOptions = new BlobUploadOptions
+            {
+                HttpHeaders = blobHttpHeaders,
+                Metadata = metadata
+            };
+
+            await sourceBlob.UploadAsync(stream, uploadOptions);
+
+            // Act
+            await destBlob.UploadFromUriAsync(sourceBlob.Uri);
```

Comment:
should we assert response ?

Reply:
We just care the the service returns a 200, otherwise a `RequestFailedException` would be thrown here.

Reply:
don't we care about parsing response correctly ?

Reply:
Fixed.